sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -89,6 +89,22 @@ object Cast {
     case _ => false
   }
 
+  def needsTimeZone(from: DataType, to: DataType): Boolean = (from, to) match {
```

Comment:
Sure. I only match on these type pairs because the `Cast` node only uses `timeZone` when doing these type conversions. I'll add that comment.